### ISSUE TITLE
[Feature][Request] sorting table by columns in related table

### ISSUE THREAD

**Issue by Jakhongir91** [Laravel-Backpack/CRUD#1893]
# Feature request

If there are two tables with 1-n relation, the opportunity to sort one table by related column in another table needed. Now we can sort by related table after page loads, pressing on column's header, also needed functionality to sort data in CrudController, so on page first load data was already sorted by related column.

The $this->crud->orderBy('column', 'order'); seems to work only for columns in current table, tried to use table name with dot:  $this->crud->orderBy('table_name.column', 'order');

**Comment by tabacitu** [comment]
Hmm this is interesting. Thank you for proposing this @Jakhongir91 - it's indeed something we should be able to do. We'll update you as we make progress.

Cheers!

**Comment by pxpm** [comment]
Hello @Jakhongir91 

I think this is something that we already provide (when I say we, I mean SQL). To be able to search for a related table you first need to add that table to the query with a join.

Something along this lines: 

```php
//imagine you are in ArticlesController and have a user_id key there
$this->crud->query->leftJoin('users', 'articles.user_id', '=', 'users.id')->orderBy('users.name','ASC');
```

You also have the `orderByRaw()` provided by Laravel, just make sure you select() the needed columns before ordering. 
 
That said, I don't think we, in Backpack need to do something for this to work.

If I am wrong, and sometimes I am, please re-open 👍 

Wish you the best,
Pedro

**Comment by tabacitu** [comment]
I totally agree @pxpm , we can do this now by joining and declaring an custom `orderByLogic` for the column. But I think we can do _more_, I think for relationship columns we can provide a _default_ `orderByLogic`, so that all the logic is provided by Backpack, and 99% of the time the dev doesn't have to do anything. They'd have magical ordering by that `relationship` column, just like it were a simple `text` column.

Not 100% sure we can achieve it in all cases in an optimal fashion, but... it should be possible to cover the 99% use case (as opposed to what we now cover - 0% 🤣 ).

So let's please reopen this, and maybe try adding a solution for it in 4.2.
